mula/scheduler/rankers/boefje.py

```diff
@@ -30,19 +34,109 @@ def rank(self, obj: Any) -> int:
         max_days_in_seconds = self.MAX_DAYS * (60 * 60 * 24)
         grace_period = timedelta(seconds=self.ctx.config.pq_grace_period)
 
+        prior_tasks: List[models.Task] = self.ctx.datastores.task_store.get_tasks_by_hash(obj.hash)
```

Comment:
Do we only want the tasks that are finished, or do we also need to count the failures?

Reply:
Lets assume we get the best measurement by looking at completed tasks. Failed tasks would skew the roi to a lower value even though it might have been a temporary issue.